src/typechecker/FStar.TypeChecker.Util.fs

```diff
@@ -2147,14 +2148,18 @@ let maybe_instantiate (env:Env.env) e t =
                       let args, bs, subst, g' = aux subst (decr_inst inst_n) rest in
                       (v, Some S.imp_tag)::args, bs, subst, Env.conj_guard g g'
 
-                  | _, (x, Some (Meta tac_or_attr))::rest ->
+                  | _, ({binder_bv=x; binder_qual=qual; binder_attrs=attrs})::rest
+                    when (match qual with | Some (Meta _) -> true
+                                          | _ -> false)
+                       || (List.length attrs > 0) ->
```

Comment:
Should arguments on explicit fields be treated the same as it's currently for implicit ones? Or rather should this look at "Implicit true && List.length attrs > 0"?
I think there are few other code pieces similar to this, I'm wondering how to update them later really.

Reply:
Yes, you are right, adding `Implicit _ && ...` is more future proof, added, thanks!